lib/Dialect/Triton/IR/Ops.cpp

```diff
@@ -145,6 +145,56 @@ void StoreOp::print(OpAsmPrinter &printer) {
 namespace mlir {
 namespace triton {
 
+//-- AddPtrOp --
+// addptr(addptr(%ptr, %idx0), %idx1) => addptr(%ptr, AddI(%idx0, %idx1))
```

Comment:
What about integer overflow? I might have `(ptr + INT_MAX) + 1` which is not the same as `ptr + (INT_MAX + 1)`.

Reply:
Good catch!